src/app/plan/plans/plans.component.ts

```diff
@@ -9,9 +10,22 @@ import { PlanService } from '../../common/services/plan.service';
 })
 export class PlansComponent implements OnInit {
   plans: Plan[];
-  constructor(private planService: PlanService) {
-    planService.getPlans().subscribe(x => this.plans = x);
+  hasPermisionsToDelete = false;
+  id: number;
+  index: number;
+  constructor(private planService: PlanService, private authService: AuthService) {
+    planService.getPlans().subscribe( (x: Plan[]) => this.plans = x);
   }
   ngOnInit() {
+    if (this.authService.isAdmin() || this.authService.isMentor()) {
+      this.hasPermisionsToDelete = true;
+    }
+  }
+
+  onDelete(id: number): void {
+    this.id = id;
+    this.planService.deletePlanById(this.id).subscribe();
+    this.index = this.plans.findIndex((plan: Plan) => plan.Id === id);
```

Comment:
Same comment with `this.index`, why store additional unused state, local variable should be enough, e.g. `let index = ...`

Reply:
ok, i'll change it